### DISCUSSION TITLE
Locale Issues Megathread

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#3619] [General]
Hey everyone,

I'll be the first to admit that the way that locales are setup isn't perfect.  There are a few inconsistencies between languages that need to be fixed.  Some smaller, some larger.

I'd like to compile a list of all of the problems that are out there.  Fixing these issues will likely end up being done all at once in their own development cycle.  I'd like this thread to be the location for all issues related to localization changed.

Note:  If the issue is about a certain wording of a translation, it may be best to have that discussion over in [Weblate](https://hosted.weblate.org/engage/free-tube/) as I won't be too helpful with that.  This thread is more for stylizing and formatting issues.

What I need from translators / native speakers is the following.

1. The language that needs changes
2. The formatting / stylized change that needs to be made
3. How it's currently formatted
4. How it needs to change
5. Anything else you feel may be helpful

Once the problem has been fixed, I will hide your post to make you aware that it's resolved.

Thanks for the help!

**Comment by milotype** [comment]
@PrestonN Should I simply link my issues/merge request here one by one in order not to repeat myself, and so you could remove them if needed? Or am I missinterpretating the intention for this thread?

**Comment by PrestonN** [comment]
Go ahead and link the issues that you've created.  Also go ahead and please close them if they are not already.

**Comment by milotype** [comment]
Alignment and typography corrections https://github.com/FreeTubeApp/FreeTube/pull/653#issuecomment-708464187 
The issue about a separator charater between "value" and "extension" remained unsolved.

**Comment by Elaborendum** [comment]
- Selecting System Default when the computer is using any Latin American Spanish (excluding, I assume, Mexican Spanish) results in the Locale not being found and the app defaulting to English. I assume this happens for other locales that are specified to a region (like maybe Belgian French?). This happened on Windows 10.
- I don't really agree with the idea of having multiple Spanish translations in projects that aren't very big, as I've seen that a lot of the time one ends up being very under worked because translators probably feel like it's tedious to do most of the same work twice. But since this project has both UK and US English, I'll assume this shall follow suit in this case. 
  - Anyway, onto the actual issue. I don't really think it's necessary to have a third Spanish translation (the Argentinian Spanish one), as the Mexican Spanish translation could cover Latin American Spanish in general, plus, it's already under worked in comparison with the other two.

Is this the right place, or should I file a separate issue for each one?

**Comment by GilgusMaximus** [comment]
Well, we are using Weblate. If the language is not translated over there, then it will not be found.

We want to keep the diversity, as we are no language experts who can tell which language differences matter and which not. E.g. Norwegian Bokmal and Nynorsk or chinese. So there are three possible solutions here:

Either changing system locale to a supported language, choosing an available language, or someone could go ahead, copy the language file, rename it according to standard to their language accent and make a PR. 

Or translate on weblate alternatively.

**Comment by Zakret** [comment]
Hi! I have an issue with this string [https://hosted.weblate.org/translate/free-tube/translations/en/?checksum=ab977b817b746e15](url). There is not enough space for the translated string. 
Polish: ![Polish](https://i.ibb.co/kM52jws/Screenshot-20210918-200941.png) Deutsche: ![Deutsche](https://i.ibb.co/c3YTLMn/Screenshot-20210918-205133.png[/img])

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @Zakret i think ur problem is actually related to https://github.com/FreeTubeApp/FreeTube/issues/1300 

Edit: @Zakret u dont have to worry about that anymore this is fixed in PR https://github.com/FreeTubeApp/FreeTube/pull/1767

**Comment by 70h** [comment]
I have completed the Lithuanian translation, but the last three versions still won't allow me to select it in the language settings. Is there something more that should be done or am I just to impatient? Thanks!

**Comment by peepo5** [comment]
Is it hidden? I do not see any lithuanian there. That is weird.

**Comment by kubo6472** [comment]
Hey there, I just updated 🇸🇰 translation. I've come around this error, not sure if it has been seen yet.
```
The following occurrences were found:
Language code | Suggested individual language codes
ku | ckb, kmr, sdh
```

**Comment by Zakret** [comment]
Hi! It looks like there are strings for the title menu on the Weblate page, but the translations are not applied

![titlebar_missing_translation](https://user-images.githubusercontent.com/78523122/174910252-94030952-eb22-4578-8076-eb026eca0756.png)

**Comment by Zakret** [comment]
There seems to be a problem with the Urdu yaml file on the Weblate:

> The translation is temporarily closed for contributions due to maintenance, please come back later.
The translation was automatically locked due to following alerts: Could not parse translation files.
static/locales/ur.yaml
'expected <block end>, but found \'<block mapping start>\' in "<unicode string>", line 132, column 5:\n General Settings: \'عام ترتیبات\'\n ^ (line: 132)'

**Comment by sveinki** [comment]
> There seems to be a problem with the Urdu yaml file on the Weblate:
> 
> > The translation is temporarily closed for contributions due to maintenance, please come back later.
> > The translation was automatically locked due to following alerts: Could not parse translation files.
> > static/locales/ur.yaml
> > 'expected , but found '' in "", line 132, column 5:\n General Settings: 'عام ترتیبات'\n ^ (line: 132)'

Meanwhile all the other translations are blocked...

**Comment by absidue** [comment]
@sveinki @Zakret We've fixed the issue in the Urdu file, hopefully weblate will unlock the languages for you soon.

**Comment by sveinki** [comment]
> @sveinki @Zakret We've fixed the issue in the Urdu file, hopefully weblate will unlock the languages for you soon.

Weblate unlocked, thanks a lot!

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/119242752/210738542-81a8101b-6f10-47ed-a379-5d5a868a3219.png)
![image](https://user-images.githubusercontent.com/119242752/210740759-22469dcc-a87d-4108-9161-cc28b965af6f.png)
The Russian translation is not displayed in the places on the attached screenshots (the player has all the buttons without translation).

**Comment by ghost** [comment]
![изображение](https://user-images.githubusercontent.com/119242752/212496566-257d7eef-8a6d-4fba-98fa-f49d69b0a539.png)

**Comment by Zakret** [comment]
Weblate seems to be locked due to parsing errors in several translation files:
[https://hosted.weblate.org/projects/free-tube/translations/#alerts](url)

> The translation is temporarily closed for contributions due to maintenance, please come back later.
The translation was automatically locked due to following alerts: Could not parse translation files.


Filename | Error
-- | --
static/locales/bs.yaml | 'could not find expected \':\'   in "<unicode string>", line 189, column 1:\n    Playlist:\n    ^ (line: 189)'
static/locales/eo.yaml | 'could not find expected \':\'   in "<unicode string>", line 153, column 1:\n    Playlist:\n    ^ (line: 153)'
static/locales/fa.yaml | 'could not find expected \':\'   in "<unicode string>", line 357, column 1:\n    Playlist:\n    ^ (line: 357)'
static/locales/fil.yaml | 'could not find expected \':\'   in "<unicode string>", line 138, column 1:\n    Playlist:\n    ^ (line: 138)'
static/locales/gsw.yaml | 'could not find expected \':\'   in "<unicode string>", line 149, column 1:\n    Playlist:\n    ^ (line: 149)'
static/locales/la.yaml | 'could not find expected \':\'   in "<unicode string>", line 211, column 1:\n    Playlist:\n    ^ (line: 211)'
static/locales/si.yaml | 'could not find expected \':\'   in "<unicode string>", line 134, column 1:\n    Playlist:\n    ^ (line: 134)'
static/locales/ti.yaml | 'could not find expected \':\'   in "<unicode string>", line 142, column 1:\n    Playlist:\n    ^ (line: 142)'
static/locales/tok.yaml | 'could not find expected \':\'   in "<unicode string>", line 147, column 1:\n    Playlist:\n    ^ (line: 147)'

**Comment by ThinksABit** [comment]
WIndows protected your PC (while trying to install Freetube) (Suggestion not criticism)

First apologies if posted in wrong location.
When trying to install this app on windows 11 windows security blocks installation with a warning.
So I do the obvious thing, run the file through virus total and it shows no issues. BUT, when i searched for any information relating to this there was nothing, anywhere, until now! May i make a suggesting addressing this small issue, in the belief that any non techy looking to try this app would instantly hit the cancel and forget about trying this software, where as a small message about this on the download page and how they could check the software is safe for themselves (to a degree, with out it be certified by MS - unknown publisher - and yes it's beta). There is a growing noise about what and how youtube does things and freetube addresses some of these issues. In doing this surly it would help the spread of use of this sotfware to a wider user base. I know this is not a strictly tech subject for many here so hope this is not inappropriate.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Hi @ThinksABit this indeed not the appropriate place for it and i will hide it after you have responded on my comment

Info u are seeking for:
https://docs.freetubeapp.io/faq/#my-virus-protection-gave-a-warning-when-i-tried-to-run-freetube-is-freetube-a-malicious-application

TLDR: FreeTube does not have a certificate and thats why smartscreen is flagging it. Acquiring a certificate costs money. We will hold off obtaining one until we reached stable v1.0.0

**Reply by ThinksABit** [reply]
Ahhh thank you. I did search in github but used different search terms. Thank you for responding and again, sorry for posting in wrong place.
